providers/env-var/src/main/java/dev/openfeature/contrib/providers/envvar/EnvVarProvider.java

```diff
@@ -0,0 +1,81 @@
+package dev.openfeature.contrib.providers.envvar;
+
+import dev.openfeature.sdk.*;
+import dev.openfeature.sdk.exceptions.ParseError;
+import dev.openfeature.sdk.exceptions.ValueNotConvertableError;
+
+import java.util.function.Function;
+
+/**
+ * EnvVarProvider is the Java provider implementation for the environment variables.
+ */
+public final class EnvVarProvider implements FeatureProvider {
+    private static final String NAME = "Environment Variables Provider";
+
+    private final OS os;
+
+    public EnvVarProvider() {
+        os = new OS();
+    }
+
+    public EnvVarProvider(OS os) {
+        this.os = os;
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx) {
+        return evaluateEnvironmentVariable(key, defaultValue, Boolean::valueOf);
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue, EvaluationContext ctx) {
+        return evaluateEnvironmentVariable(key, defaultValue, string -> string);
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue, EvaluationContext ctx) {
+        return evaluateEnvironmentVariable(key, defaultValue, Integer::valueOf);
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue, EvaluationContext ctx) {
+        return evaluateEnvironmentVariable(key, defaultValue, Double::valueOf);
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        throw new ValueNotConvertableError("EnvVarProvider supports only primitives");
+    }
+
+    private <T> ProviderEvaluation<T> evaluateEnvironmentVariable(
+            String key,
+            T defaultValue,
+            Function<String, T> parse
+    ) {
+        final String value = os.getenv(key);
```

Comment:
One thing we've done in other ENV VAR providers is to map `SCREAMING_SNAKE_CASE` env vars to `hyphen-case-keys`. This _may_ be desirable, since generally people like to use `SCREAMIN_SNAKE` for env vars, but that's generally not true for feature flag keys.

You could also create some transformer interface to allow users to control this translation.... and it doesn't have to be done in this PR.